assets/js/admin-settings/index.jsx

```diff
@@ -0,0 +1,118 @@
+/* global localizedData */
+import {
+  Button,
+  SelectControl,
+  TextControl,
+  ToggleControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import React, { useEffect, useState } from 'react';
+import useSiteOptions from '../services/hooks/use-site-options';
+import Rule from './rule';
+import { ruleCorral } from './styles';
+
+
+const AdminSettings = () => {
+  const [{ loading, saving, settings }, setOptions] = useSiteOptions();
+  const [originIndex, setOriginIndex] = useState(null);
+  const [targetIndex, setTargetIndex] = useState(null);
+
+  /**
+   * Helper function for saving the in-memory settings to the server.
+   */
+   const saveSettings = (updatedRules) => {
+    const next = { ...settings, apple_news_automation: updatedRules };
+
+    // Enforce some defaults prior to save.
+    // Request will 500 when site_logo === null.
+    next.site_logo = next.site_logo ?? 0;
+    console.log('settings-pre-save', next);
+
+    // Kick off the save to the server.
+    setOptions(next);
+  };
+  
+  const cleanData = () => {
+    saveSettings([]);
+  }
+  
+  // useEffect(() => {
+  //   cleanData();
+  // },[]);
+
+  const addRule = (newRule) => {
+    const updatedRules = settings.apple_news_automation ?? [];
+    updatedRules.push(newRule);
+    saveSettings(updatedRules);
```

Comment:
I would recommend _not_ calling `saveSettings` until the user clicks a Save button. Let's keep all changes local until a save is requested.

Reply:
Actually, we might want to refactor `useSiteOptions` to hold the settings in state and have a separate method for saving the data back to WordPress using the API - that would keep the work here more lightweight.